saleor/payment/utils.py

```diff
@@ -130,21 +131,38 @@ def gateway_charge(
     clean_capture(payment, amount)
 
     gateway, gateway_params = get_payment_gateway(payment.gateway)
-    with transaction.atomic():
-        txn, error = gateway.charge(
+    try:
+        is_success, gateway_response, error = gateway.charge(
```

Comment:
I have a few concerns/suggestions for this:
- IMHO, for this ```gateway.charge``` as well as others ```gateway.authorize|capture|etc```, a wrapper for them can be added. This wrapper or layer can check whether the gateway is returning the correct format. With Saleor's payment evolves, the expected format of a gateway could be changed, and the third-party gateway repo could be outdated. It will be nice to check the gateway response before passing the gateway response to saleor. If the gateway response format is out-of-date, raise exceptions or warnings. This can solve a lot of issues in advance.
- This layer could also used for deprecation cycles. For example, if saleor payment is going to deprecate certain field from the gateway response such as the ```is_success``` field, the deprecation warning can done in this wrapper. 
- Returning ```is_success, gateway_response, error``` probably is not the best way to get the gateway response. Maybe a dict or namedtuple? So extra result could be passed back to saleor without raising errors. 
 Otherwise one minor extra return may break the whole payment system.
- Since ```gateway_response``` is used directly to create transactions which means this is not the raw gateway (I mean paypal, stripe or whatever as gateway here) response. The transactions in saleor should used saleor's standard of ```amount``` or ```currency```, but the raw gateway response may not have these kind of information directly. Some gateway may not return ```currency``` at all. So this ```gateway_response``` is processed response from the providers (I means saleor-payment-paypal or saleor-payment-stripe and whatever here as providers), what is the standard of the gateway response? Which fields must it include? It will be nice to define it.
- It will be nice to add a new field such as ```raw_gateway_response``` to ```Transaction``` model. This field could save the raw gateway response. This helps debugging. Sometimes due to bugs, the postprocessed ```gateway_response``` could be the wrong data.